### DISCUSSION TITLE
Ubuntu - undefined reference to PxSetPhysXGpuProfilerCallback

### DISCUSSION THREAD

**Discussion by Kai-Indiemade** [NVIDIA-Omniverse/PhysX#135] [Q&A]
I seem to have the libraries(.lib files) successfully linked on Linux via Cmake, but I am getting the following error:
PxPvdImpl.cpp:108: undefined reference to 'PxSetPhysXGpuProfilerCallback'

It appears it is trying to include code that is for the PhysXGpu library, which to my knowledge is supposed to optional? It it is not, how would I link it to my project? I tried linking the dll, but that caused other undefined references, so I'm assuming PhysXGpu has some dependencies of it's own.

I found a preprocessor directive "DISABLE_CUDA_PHYSX" that looks like it'd avoid the calls to PhysXGpu, but defining it didn't seem to make a difference.

**Comment by preist-nvidia** [comment]
You need to unset `PX_SUPPORT_GPU_PHYSX`.

See where you hit the issue: https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.2/physx/source/pvd/src/PxPvdImpl.cpp#L105

**Comment by Kai-Indiemade** [comment]
Hi there, I tried that with:
#undef PX_SUPPORT_GPU_PHYSX

However, that still did not work. I unset it before I include any PhysX headers. Let me know if you need any more info, this issue does not seem to happen on Windows.(And on Windows, there is not any need to define/undefine any GPU related preprocessor directives.)

**Comment by StuAtGit** [comment]
@Kai-Indiemade - unsetting before including any PhysX headers won't do anything, as it's a macro defined in the PhysX codebase. That is, it's going to effect how _PhysX_ compiles. 

You'll need to recompile PhysX with the macro undefined.

I'm starting with updating the presets XML file (`buildtools/presets/public/linux.xml`) to set it to `False`, and doing a `make clean && make` in the `checked` dir.

If that fails, 
  - I'll try setting `DISABLE_CUDA_PHYSX` in the presets to true
  - And if that fails, I'll try directly updating the preprocessor directives in `include/foundation/PxPreprocessor.h` a go:

```
//#define DISABLE_CUDA_PHYSX
#ifndef DISABLE_CUDA_PHYSX
	//CUDA is currently supported on x86_64 windows and linux, and ARM_64 linux
	#define PX_SUPPORT_GPU_PHYSX ((PX_X64 && (PX_WINDOWS_FAMILY || PX_LINUX)) || (PX_A64 && PX_LINUX))
#else
	#define PX_SUPPORT_GPU_PHYSX 0
#endif
```

I'll update if it works - thanks for asking the question, I never would have found this without it!

**Comment by StuAtGit** [comment]
So, you end up needing to edit `PxPreprocessor.h` to disable CUDA.

However, with 105.1-physx-5.3.0

that just leads you to this issue:

https://github.com/NVIDIA-Omniverse/PhysX/issues/196

the fix in the issue (void)param will get you past the errors in softbody, but that just ends up cascading down to more errors in other files.

Attempting to compile `main` with CUDA disabled now... :crossed_fingers:

**Comment by StuAtGit** [comment]
Not fixed in `main`, it would seem. Must be living in some code / branch not available to the public, as nvidia marked the issue as `completed` and stated it would be fixed in the next release.

**Reply by preist-nvidia** [reply]
Yes, the 196 fix is not in main yet until the next release.

**Comment by StuAtGit** [comment]
Update: appear to be fixed in main. I did have to disable snippet to get it to compile. That appears to be an unrelated (new) issue, though.
